File: GenshinTool-SPA/src/app/Genshin/models/Artefact/artefact.model.ts
import { Stat } from "../Stat/stat.model";
import { ArtefactSet } from "./ArtefactSet.model";
import { ArtefactPiece } from "./artefactPiece.model";

export class Artefact{
    id!: number;
    setId!: number;
    pieceId!: number;
    set!: ArtefactSet;
    piece!: ArtefactPiece;
    stats!: Stat[];
}

File: GenshinTool.Common/Models/Domain/Concretes/ArtefactModel.cs
using GenshinTool.Common.Models.Core.Model;
using GenshinTool.Common.Models.Domain.Interfaces;
using System.Text.Json.Serialization;

namespace GenshinTool.Common.Models.Domain.Concretes;

public class ArtefactModel : BaseModel, IArtefactModel
{
    public long SetId { get; set; }
    public long PieceId { get; set; }
    public ArtefactPieceModel? Piece { get; set; }
    public ArtefactSetModel? Set { get; set; }
    public IEnumerable<StatModel>? Stats { get; set; }
}

